akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala

```diff
@@ -78,8 +78,10 @@ private[remote] class ArteryTcpTransport(
   // may change when inbound streams are restarted
   @volatile private var inboundKillSwitch: SharedKillSwitch = KillSwitches.shared("inboundKillSwitch")
   // may change when inbound streams are restarted
-  @volatile private var inboundStream: OptionVal[Sink[EnvelopeBuffer, NotUsed]] = OptionVal.None
   @volatile private var serverBinding: Option[ServerBinding] = None
+  private val firstConnectionFlow = Promise[Flow[ByteString, ByteString, NotUsed]]()
+  @volatile private var inboundConnectionFlow: Future[Flow[ByteString, ByteString, NotUsed]] =
```

Comment:
So, the potential race condition that could happen here would be that we have already bound the port in `bindInboundStreams` and there's already an incoming connection on it before we have run `runInboundStreams`?

Reply:
Correct